### PR TITLE
Downloader interface

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: include/aos/common/downloader.hpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+     *
+     * @param url URL.
+     * @param path path to file.
+     * @param contentType content type.
+     * @return Error.
+     */
+    virtual Error Download(const aos::String& url, const aos::String& path, const aos::String& contentType) = 0;
+
+    /**
+     * @brief Destroy the Downloader Itf object
```
Comment (al1img) [inline include/aos/common/downloader.hpp:30]: No need for @brief tag

File: include/aos/common/downloader.hpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+#ifndef AOS_DOWNLOADER_HPP_
+#define AOS_DOWNLOADER_HPP_
+
+#include "aos/common/error.hpp"
+#include "aos/common/string.hpp"
+
+namespace aos {
+namespace downloader {
```
Comment (al1img) [inline include/aos/common/downloader.hpp:15]: All common instances just in aos namespace.

File: include/aos/common/downloader.hpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+class DownloaderItf {
```
Comment (al1img) [inline include/aos/common/downloader.hpp:17]: Add comment for class itself as well.

File: include/aos/common/downloader.hpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+public:
+    /**
+     * Downloads file.
+     *
+     * @param url URL.
+     * @param path path to file.
+     * @param contentType content type.
+     * @return Error.
+     */
+    virtual Error Download(const aos::String& url, const aos::String& path, const aos::String& contentType) = 0;
```
Comment (al1img) [inline include/aos/common/downloader.hpp:27]: I forget, do we need content type? Could we use aos::EnumStringer for it?

File: include/aos/common/downloader.hpp [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/*
```
Comment (al1img) [inline include/aos/common/downloader.hpp:1]: add tag [donwloader] to commit message.

File: include/aos/common/downloader.hpp (hunk 1 above)
Comment (MykolaSolyanko) [reply include/aos/common/downloader.hpp:30]: Done

File: include/aos/common/downloader.hpp (hunk 2 above)
Comment (MykolaSolyanko) [reply include/aos/common/downloader.hpp:15]: Done

File: include/aos/common/downloader.hpp (hunk 3 above)
Comment (MykolaSolyanko) [reply include/aos/common/downloader.hpp:17]: Done

File: include/aos/common/downloader.hpp (hunk 4 above)
Comment (MykolaSolyanko) [reply include/aos/common/downloader.hpp:27]: Done

File: include/aos/common/downloader.hpp [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+#define AOS_DOWNLOADER_HPP_
+
+#include "aos/common/enum.hpp"
+#include "aos/common/error.hpp"
+#include "aos/common/string.hpp"
+
+namespace aos {
+
+/**
+ * Instance run state.
```
Comment (al1img) [inline include/aos/common/downloader.hpp:18]: Wrong comment.

File: include/aos/common/downloader.hpp [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+    static const Array<const char* const> GetStrings()
+    {
+        static const char* const cContentTypeStrings[] = {"service"};
+
+        return Array<const char* const>(cContentTypeStrings, ArraySize(cContentTypeStrings));
+    };
+};
+
+using DownloadContentTypeEnum = DownloadContentType::Enum;
```
Comment (al1img) [inline include/aos/common/downloader.hpp:32]: `DownloadContentEnum`

File: include/aos/common/downloader.hpp [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+using ContentType = EnumStringer<DownloadContentType>;
```
Comment (al1img) [inline include/aos/common/downloader.hpp:33]: `DownloadContent`

File: include/aos/common/downloader.hpp [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Downloader interface
```
Comment (al1img) [inline include/aos/common/downloader.hpp:36]: period at the end.

File: include/aos/common/downloader.hpp [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+     *
+     * @param url URL.
+     * @param path path to file.
+     * @param contentType content type.
+     * @return Error.
+     */
+    virtual Error Download(const String& url, const String& path, ContentType contentType) = 0;
+
+    /**
+     * Destroy the Downloader Itf object
```
Comment (al1img) [inline include/aos/common/downloader.hpp:51]: Destroys.

File: include/aos/common/types.hpp [hunk 11]
```diff
@@ -77,6 +77,11 @@ constexpr auto cSHA256Size = 32;
  */
 constexpr auto cSHA512Size = 64;
 
+/**
+ * Error len.
+ */
+constexpr auto cErrorLen = 256;
```
Comment (al1img) [inline include/aos/common/types.hpp:83]: `cErrorMessageLen`?

File: include/aos/common/downloader.hpp (hunk 6 above)
Comment (MykolaSolyanko) [reply include/aos/common/downloader.hpp]: Done

File: include/aos/common/downloader.hpp (hunk 7 above)
Comment (MykolaSolyanko) [reply include/aos/common/downloader.hpp]: Done

File: include/aos/common/downloader.hpp (hunk 8 above)
Comment (MykolaSolyanko) [reply include/aos/common/downloader.hpp]: Done

File: include/aos/common/downloader.hpp (hunk 9 above)
Comment (MykolaSolyanko) [reply include/aos/common/downloader.hpp]: Done

File: include/aos/common/downloader.hpp (hunk 10 above)
Comment (MykolaSolyanko) [reply include/aos/common/downloader.hpp]: Done

**Review by al1img — APPROVED** [review]

**Pull request merged**